assets/js/blocks/cart-checkout-shared/payment-methods/payment-method-error-boundary.tsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useState } from '@wordpress/element';
+import { CURRENT_USER_IS_ADMIN } from '@woocommerce/settings';
+import { StoreNoticesContainer } from '@woocommerce/blocks-checkout';
+import { noticeContexts } from '@woocommerce/base-context';
+import { NoticeType } from '@woocommerce/types';
+interface PaymentMethodErrorBoundaryProps {
+	isEditor: boolean;
+	children: React.ReactNode;
+}
+const PaymentMethodErrorBoundary = ( {
+	isEditor,
+	children,
+}: PaymentMethodErrorBoundaryProps ) => {
+	const [ errorMessage ] = useState( '' );
+	const [ hasError ] = useState( false );
+	if ( hasError ) {
+		let errorText = __(
+			'We are experiencing difficulties with this payment method. Please contact us for assistance.',
+			'woo-gutenberg-products-block'
+		);
+		if ( isEditor || CURRENT_USER_IS_ADMIN ) {
+			if ( errorMessage ) {
+				errorText = errorMessage;
+			} else {
+				errorText = __(
+					"There was an error with this payment method. Please verify it's configured correctly.",
+					'woo-gutenberg-products-block'
+				);
+			}
+		}
+		const notices: NoticeType[] = [
+			{
+				id: '0',
+				content: errorText,
+				isDismissible: false,
+				status: 'error',
+			},
+		];
+		return (
+			<StoreNoticesContainer
+				additionalNotices={ notices }
+				context={ noticeContexts.PAYMENTS }
+			/>
+		);
+	}
+	return <>{ children }</>;
```

Comment:
We can just return children here like with the original code. 
```suggestion
	return children;
```

Reply:
Hi @mikejolley, I returned an element as we are using PaymentMethodErrorBoundary as JSX.Element and it gives
`'PaymentMethodErrorBoundary' cannot be used as a JSX component.
  Its return type 'string | number | boolean | Element | ReactFragment | null | undefined' is not a valid JSX element.
    Type 'undefined' is not assignable to type 'Element | null'.ts(2786)` 
    
    any feedback on this?

Reply:
Thats because at the top the types are defined as:

```
children: React.ReactNode;
```

I think this could be changed to:

```
children: JSX.Element | JSX.Element[];
```

That should fix it?

Reply:
Hi @mikejolley yes it was `children: JSX.Element | JSX.Element[];` before but it was causing the following error
![image](https://github.com/woocommerce/woocommerce-blocks/assets/33057454/bb7b2fea-7268-4ef7-9500-db9ae96567a9)
So then I again made `childered: JSX.Element`
which causes
![image](https://github.com/woocommerce/woocommerce-blocks/assets/33057454/72a0f6cf-9c64-4d0a-ac34-d14dda0a9943)
went in circle...

Is any of the above acceptable, ignoring the ts error..
Is there anything else I can try? or somewhere I can take some reference from?

Reply:
Ah good catch. I think we should update the type instead of the return value. How about this:

```
import type { Children } from 'react';
interface PaymentMethodErrorBoundaryProps {
	isEditor: boolean;
	children: typeof Children;
}
```

We could have used `React.ReactChildren` but its deprecated in favour of the above. cc @opr do you agree?

Reply:
Yes, just tried that out, it seems to work fine! Nice suggestion

Reply:
@hritikchaudhary are you ok to update this? Thats the last thing blocking merge.

Reply:
@mikejolley Yes sure, np

Reply:
Hi @opr I tied this out but unsuccessfully can you please check if I'm missing something
<img width="424" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/33057454/d525851a-3d50-4ed1-af20-fcd58a9b8306">
<img width="424" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/33057454/35fe1510-0ce4-43ce-83cb-cab0d12804b4">

But it did not resolve the second error in `PaymentMethodCard` it got changed to 
`This JSX tag's 'children' prop expects a single child of type '{ map<T, C>(children: C | readonly C[], fn: (child: C, index: number) => T): C extends null | undefined ? C : Exclude<T, boolean | null | undefined>[]; forEach<C>(children: C | readonly C[], fn: (child: C, index: number) => void): void; count(children: any): number; only<C>(children: C): C extends any[] ? never : C;...', but multiple children were provided.ts(2746)`

<img width="599" alt="image" src="https://github.com/woocommerce/woocommerce-blocks/assets/33057454/bf8dc49a-3722-4c5e-b6dc-590397e6feff">

Reply:
Maybe wrap that second one with `<></>`; looks like a common issue with conditionals https://stackoverflow.com/questions/62382324/react-typescript-this-jsx-tags-children-prop-expects-a-single-child-of-type

Reply:
Do you mean wrapping this [line](https://github.com/woocommerce/woocommerce-blocks/pull/9817/files#diff-61c5e62d8c51dd6baf84622047dd73c96473296eefc52733c45f2408fb6a7023R50)
I did not push the commit so it is already returning 
`return <>{ children }</>;`

Reply:
yeah whatever solves the warning. I'd still like to see the switch to `import type { Children } from 'react'` however, as the type is more accurate.

Reply:
ok

Reply:
Hi @mikejolley, Apologies for delay... I worked on it today and did some trial and error but only combination that worked for me is using `children: React.ReactNode;` and returning `return <>{ children }</>;`.
using `import type { Children } from 'react'` gives `This JSX tag's 'children' prop expects a single child of type...` error.
That's why I didn't push any changes..

Reply:
Ok lets leave it.